object.go

```diff
@@ -430,15 +448,15 @@ func (obs *obs) Put(meta *ObjectMeta, r io.Reader, opts ...ObjectOpt) (*ObjectIn
 	case <-time.After(obs.js.opts.wait):
 		return nil, ErrTimeout
 	}
-	info.ModTime = time.Now().UTC()
 
-	// Delete any original one.
+	// Delete any original chunks.
 	if einfo != nil && !einfo.Deleted {
-		chunkSubj := fmt.Sprintf(objChunksPreTmpl, obs.name, einfo.NUID)
-		obs.js.purgeStream(obs.stream, &streamPurgeRequest{Subject: chunkSubj})
+		purgeChunks(obs, einfo.NUID)
 	}
 
-	return info, nil
+	// Yes this is an extra call, but GetInfo sets the modified time correctly
+	// to the server message time, instead of just setting the time based on the client
+	return obs.GetInfo(meta.Name)
```

Comment:
Derek - the original code just set the modified time on the info after it was publish, which isn't really the actual message timestamp. If we don't mind the extra server round trip here, then leave it, otherwise I would prefer leaving the modified time nil instead of giving it time from the client.

Reply:
I was trying to use modified time as a signal, but would need to dig in further.